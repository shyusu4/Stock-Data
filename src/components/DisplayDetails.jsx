import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import { dispatchCompanies } from '../redux/stockData';
import DetailsHeader from './DetailsHeader';

const DisplayDetails = () => {
    const dispatch = useDispatch();
    const company = useSelector((state) => state.stockReducer.company || '',);
    const { name } = useParams();
    useEffect(() => {
      dispatch(dispatchCompanies(name));
    }, [company.name, dispatch, name]);
  
  return (
    <>
    <DetailsHeader />
    <div className='details-container'>
      <div className='details-header'>
        <div className='details-img-div'><img src={company.image} alt="company logo" className='details-img img-fluid'/></div>
        <p className="stats-breakdown">Stats Breakdown</p>
      </div>
      <div className='details-body'>
<div className='details-metric'>
  <p>Company Name: {company.companyName}</p>
</div>
<div className='details-metric'>
  <p>Symbol: {company.symbol}</p>
</div>
<div className='details-metric'>
  <p>Currency: {company.currency}</p>
</div>
<div className='details-metric'>
  <p>Exchange: {company.exchange}</p>
</div>
<div className='details-metric'>
  <p>Price: {company.price}</p>
</div>
<div className='details-metric'>
  <p>Range: {company.range}</p>
</div>
<div className='details-metric'>
  <p>Changes: {company.changes}</p>
</div>
<div className='details-metric'>
  <p>Market Cap: {company.mktCap}</p>
</div>
      </div>
    </div>
    </>
  );
}

export default DisplayDetails;