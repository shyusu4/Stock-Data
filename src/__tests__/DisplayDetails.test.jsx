import React from 'react';
import Renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import DisplayDetails from '../components/DisplayDetails';

// eslint-disable-next-line func-names
window.matchMedia = window.matchMedia || function () {
  return {
    matches: false,
    addListener() {},
    removeListener() {},
  };
};

describe('DisplayDetails', () => {
  it('Check if DisplayDetails component renders correctly', () => {
    const tree = Renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <DisplayDetails />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
