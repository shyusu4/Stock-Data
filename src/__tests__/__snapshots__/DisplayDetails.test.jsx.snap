// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayDetails Check if DisplayDetails component renders correctly 1`] = `
Array [
  <nav
    className="mb-3 fixed-top navbar navbar navbar-expand-lg navbar-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-name navbar-brand"
        href="#"
      >
        Stock Market Metrics
      </a>
      <a
        href="/"
        onClick={[Function]}
      >
        <svg
          className="text-white fw-bolder fs-3"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zm-4.5-.5a.5.5 0 0 1 0 1H5.707l2.147 2.146a.5.5 0 0 1-.708.708l-3-3a.5.5 0 0 1 0-.708l3-3a.5.5 0 1 1 .708.708L5.707 7.5H11.5z"
            fillRule="evenodd"
          />
        </svg>
      </a>
    </div>
  </nav>,
  <div
    className="details-container"
  >
    <div
      className="details-header"
    >
      <div
        className="details-img-div"
      >
        <img
          alt="company logo"
          className="details-img img-fluid"
        />
      </div>
      <p
        className="stats-breakdown"
      >
        Stats Breakdown
      </p>
    </div>
    <div
      className="details-body"
    >
      <div
        className="details-metric"
      >
        <p>
          Company Name:
           
        </p>
      </div>
      <div
        className="details-metric"
      >
        <p>
          Symbol:
           
        </p>
      </div>
      <div
        className="details-metric"
      >
        <p>
          Currency:
           
        </p>
      </div>
      <div
        className="details-metric"
      >
        <p>
          Exchange:
           
        </p>
      </div>
      <div
        className="details-metric"
      >
        <p>
          Price:
           
        </p>
      </div>
      <div
        className="details-metric"
      >
        <p>
          Range:
           
        </p>
      </div>
      <div
        className="details-metric"
      >
        <p>
          Changes:
           
        </p>
      </div>
      <div
        className="details-metric"
      >
        <p>
          Market Cap:
           
        </p>
      </div>
    </div>
  </div>,
]
`;
